turbinia/workers/analysis/macho.py

```diff
@@ -0,0 +1,600 @@
+            cd_ident = code_directory[cd_ident_offset:-1].split(
+                b'\0')[0].decode()
+            signature.identifier = cd_ident
+          if cd_team_id_offset > 0:
+            cd_team_id = code_directory[cd_team_id_offset:-1].split(
+                b'\0')[0].decode()
+            signature.team_identifier = cd_team_id
+        elif generic_blob_magic == self._CSMAGIC_BLOBWRAPPER and blob_index_type == self._CSSLOT_SIGNATURESLOT:
+          result.log(
+              f'     found CSMAGIC_BLOBWRAPPER (0xfade0b01) with CMS Signature slot'
+          )
+          signature.size = generic_blob_length
+          blobwrapper_base = blob_index_offset + 8
+          cert = signature_bytes[blobwrapper_base:blobwrapper_base +
+                                 generic_blob_length]
+          content_info = cms.ContentInfo.load(cert)
+          if content_info['content_type'].native == 'signed_data':
+            signed_data = content_info['content']
+            signer_infos = signed_data['signer_infos']
+            for signer_info in signer_infos:
+              signer = SignerInfo()
+              signed_attrs = signer_info['signed_attrs']
+              for signed_attr in signed_attrs:
+                signed_attr_type = signed_attr['type']
+                signed_attr_values = signed_attr['values']
+                if signed_attr_type.native == 'signing_time':
+                  signer.signing_time = str(signed_attr_values.native[0])
+                elif signed_attr_type.native == 'message_digest':
+                  # https://forums.developer.apple.com/forums/thread/702351
+                  signer.message_digest = signed_attr_values.native[0].hex()
+                  if len(signed_attr_values.native[0]) > 20:
+                    signer.cd_hash = signed_attr_values.native[0][0:20].hex()
+                  else:
+                    signer.cd_hash = signed_attr_values.native[0].hex()
+              sid = signer_info['sid']
+              issuer = sid.chosen['issuer'].chosen
+              for entry in issuer:
+                # https://datatracker.ietf.org/doc/html/rfc5652#section-10.2.4
+                for sub_entry in entry:
+                  # https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4
+                  name_type = sub_entry['type']
+                  val = sub_entry['value']
+                  if name_type.native == 'country_name':
+                    signer.country_name = val.native
+                  elif name_type.native == 'common_name':
+                    signer.common_name = val.native
+                  elif name_type.native == 'organization_name':
+                    signer.organization_name = val.native
+                  elif name_type.native == 'organizational_unit_name':
+                    signer.organizational_unit_name = val.native
+              signature.signer_infos.append(signer)
+    else:
+      result.log(f'*** no embedded code signature detected ***')
```

Comment:
There's a few log lines with various formats of prefix/postfix characters (dashes, asterisks, and equals signs) and capitalization.  This is going to a normal log file, so usually we don't have much other markup data here, but I can see how it might be useful in this case when logging lots of structured data.   Can you send me the output of what this looks like?  I wondering if we can make the formatting a bit more consistent.

Reply:
Good point. I will clean up and unify the output to match what we do in the rest of the code currently and remove  logging lines that are no longer needed.